backend/oqtopus_cloud/provider/routers/jobs.py

```diff
@@ -39,42 +40,73 @@
 
 @router.get(
     "/jobs",
-    response_model=list[JobDef],
+    response_model=list[Union[JobDef, GetJobsResponse]],
+    responses={500: {"model": Detail}},
 )
 @tracer.capture_method
 def get_jobs(
     device_id: str,
+    fields: Optional[str] = None,
     status: Optional[str] = None,
     max_results: Optional[int] = None,
     timestamp: Optional[str] = None,
     db: Session = Depends(get_db),
-) -> list[JobDef]:
+) -> list[Union[JobDef, GetJobsResponse]] | ErrorResponse:
     logger.info("invoked get_jobs")
-    query = db.query(Job).filter(Job.device_id == device_id)
-    if status is not None:
-        query = query.filter(Job.status == status)
-    if timestamp is not None:
-        time = datetime.fromisoformat(timestamp).astimezone(jst)
-        query = query.filter(Job.created_at > time)
-    if max_results is not None:
-        query = query.limit(max_results)
-    models = query.all()
-    jobs: list[JobDef] = []
-    for model in models:
-        job = model_to_schema(model)
-        if isinstance(job, ValueError):
-            logger.warning(str(job))
+    # Fields Control
+    try:
+        if fields is not None:
+            fields_list = fields.split(",")
+            MAP_SCHEMA_TO_MODEL = {v: k for k, v in MAP_MODEL_TO_SCHEMA.items()}
+            converted_fields_list = [
+                MAP_SCHEMA_TO_MODEL[field] for field in fields_list
```

Comment:
I think `fields` parameter must be validated (i.e. make sure that every field is a member of `Job`) before given to the `MAP_SCHEMA_TO_MODEL`, or it would end up with `KeyError` and returning internal server error to user.

Reply:
https://github.com/oqtopus-team/oqtopus-cloud/blob/95b0788414394d812526f434c26ecd95285cd793/backend/oqtopus_cloud/user/routers/jobs.py#L86-L97
Add validation for `fields` parameter.